### PR TITLE
[PWA-1033] Localize additional strings

### PR DESCRIPTION
**Pull request by anthoula**
## Description

Localize remaining client-side strings from newly delivered features, as well as from strings found using the new `jsx-no-literals` ESLint rule

## Related Issue

https://jira.corp.magento.com/browse/PWA-1033

### Verification Stakeholders

@sirugh 
@dpatil-magento 
### Verification Steps

1. Go areas of the app where code changes occurred.
- NotFound
- Account Information Page
- App
- Category List
- Checkout Page
- CMS Block
- My Account, Reset Password
- Order History Page
- Wishlist Page, Create Wishlist
2. Verify strings are localized.

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/components/App/__tests__/__snapshots__/app.spec.js.snap [hunk 1]
```diff
@@ -740,8 +740,7 @@ ReactTestInstance {
                     }
                     src={[Function]}
                   />,
-                  "message": "Sorry! An unexpected error occurred.
-Debug: 1 1",
+                  "message": "Sorry! An unexpected error occurred.\\\\nDebug: 1 1",
```
Comment (sirugh) [inline packages/venia-ui/lib/components/App/__tests__/__snapshots__/app.spec.js.snap:743]: Are these `\\\` expected?

File: packages/venia-ui/lib/components/App/app.js [hunk 2]
```diff
@@ -23,36 +24,50 @@ const OnlineIcon = <Icon src={WifiIcon} attrs={{ width: 18 }} />;
...
-    }, [addToast]);
+    }, [addToast, formatMessage]);
 
     const handleError = useCallback(
         (error, id, loc, handleDismissError) => {
             const errorToastProps = {
                 icon: ErrorIcon,
-                message: `${ERROR_MESSAGE}\nDebug: ${id} ${loc}`,
+                message: formatMessage(
+                    { id: 'app.errorDebug', defaultMessage: '' },
```
Comment (sirugh) [inline packages/venia-ui/lib/components/App/app.js:68]: If we do keep this, the default should at least be `ERROR_MESSAGE`

File: packages/venia-ui/lib/components/CategoryList/categoryList.js [hunk 3]
```diff
@@ -48,7 +49,11 @@ const CategoryList = props => {
     if (error) {
         child = (
             <div className={classes.fetchError}>
-                Data Fetch Error: <pre>{error.message}</pre>
+                <FormattedMessage
+                    id={'categoryList.errorFetch'}
+                    defaultMessage={'Data Fetch Error:'}
```
Comment (sirugh) [inline packages/venia-ui/lib/components/CategoryList/categoryList.js:54]:
```suggestion
                    defaultMessage={'Data Fetch Error: '}
```

File: packages/venia-ui/i18n/en_US.json [hunk 4]
```diff
@@ -38,6 +47,8 @@
     "categoryContent.filter": "Filter",
     "categoryContent.itemsSortedBy": "Items sorted by ",
     "categoryLeaf.allLabel": "All {name}",
+    "categoryList.errorFetch": "Data Fetch Error:",
```
Comment (sirugh) [inline packages/venia-ui/i18n/en_US.json:50]:
```suggestion
    "categoryList.errorFetch": "Data Fetch Error: ",
```

File: packages/venia-ui/i18n/en_US.json [hunk 5]
```diff
@@ -108,6 +122,9 @@
     "createAccount.passwordText": "Password",
     "createAccount.subscribeText": "Subscribe to news and updates",
     "createWishlist.createListText": "Create a list",
+    "createWishlist.dialogTitle": "New Favorites List",
+    "createWishlist.handleCreateListText": "Create a list",
+    "createWishlist.listName": "List Name",
```
Comment (sirugh) [inline packages/venia-ui/i18n/en_US.json:127]: Where did these come from?

**Review by sirugh — CHANGES_REQUESTED** [review]

File: packages/venia-ui/lib/components/CategoryList/categoryList.js (hunk 3 above)
Comment (sirugh) [reply packages/venia-ui/lib/components/CategoryList/categoryList.js:54]: We didn't have snaps for this before, but I think the JSX would have rendered the space between the string and the `<pre>`

File: packages/venia-ui/i18n/en_US.json (hunk 5 above)
Comment (anthoula) [reply packages/venia-ui/i18n/en_US.json:127]: There were console warnings complaining about missing translations, so they've been added. Traced it back to a feature merge ~6 days ago that did not have these included.

**Review by sirugh — APPROVED** [review]

**Pull request merged**